src/components/schedule-visualizer/robot.tsx

```diff
@@ -34,7 +42,17 @@ const Robot = React.forwardRef(function(
 ): React.ReactElement {
   const resourcesContext = useContext(ResourcesContext);
   const classes = useStyles();
-  const { robot, footprint, colorManager, fleetName, inConflict, onClick } = props;
+  const {
+    robot,
+    footprint,
+    colorManager,
+    fleetName,
+    inConflict,
+    NameLabelComponent,
+    onClick,
+  } = props;
+  const Text = NameLabelComponent || SvgText;
```

Comment:
Can we use something like useMemo here? 

I ran some performance test with React Profiler and I got these numbers

Current:
![image](https://user-images.githubusercontent.com/11761240/91750883-d4ba8400-eb91-11ea-88cb-c3c7870b830b.png)

With useMemo
![image](https://user-images.githubusercontent.com/11761240/91751141-44c90a00-eb92-11ea-9bf9-da510f739919.png)

HTH

Reply:
I removed NameLabelComponent prop now that `SvgText` can work in jsdom. Changed `RobotComponent` in `RobotsOverlay` to use memo as suggested, I'm surprised to see so much impact in a || statement.